src/constants/chains.ts

```diff
@@ -1,90 +1,57 @@
-/**
- * SupportedChainId must be defined inline, without using @uniswap/sdk-core, so that its members are their own types
- * {@see https://www.typescriptlang.org/docs/handbook/enums.html#union-enums-and-enum-member-types}. This allows the
- * derived const arrays and their types (eg {@link L1_CHAIN_IDS}, {@link SupportedL1ChainId}) to be narrowed and used
- * to enforce chain typing.
- *
- * Because this is not explicitly derived from @uniswap/sdk-core, there is a unit test to enforce conformance.
- */
-export enum SupportedChainId {
-  MAINNET = 1,
-  GOERLI = 5,
-  SEPOLIA = 11155111,
-
-  ARBITRUM_ONE = 42161,
-  ARBITRUM_GOERLI = 421613,
+import { ChainId, SUPPORTED_CHAINS, SupportedChainsType } from '@uniswap/sdk-core'
 
-  OPTIMISM = 10,
-  OPTIMISM_GOERLI = 420,
-
-  POLYGON = 137,
-  POLYGON_MUMBAI = 80001,
-
-  CELO = 42220,
-  CELO_ALFAJORES = 44787,
-
-  BNB = 56,
-}
-
-export const UniWalletSupportedChains = [
-  SupportedChainId.MAINNET,
-  SupportedChainId.ARBITRUM_ONE,
-  SupportedChainId.OPTIMISM,
-  SupportedChainId.POLYGON,
-]
+export const UniWalletSupportedChains = [ChainId.MAINNET, ChainId.ARBITRUM_ONE, ChainId.OPTIMISM, ChainId.POLYGON]
 
 export const CHAIN_IDS_TO_NAMES = {
-  [SupportedChainId.MAINNET]: 'mainnet',
-  [SupportedChainId.GOERLI]: 'goerli',
-  [SupportedChainId.SEPOLIA]: 'sepolia',
-  [SupportedChainId.POLYGON]: 'polygon',
-  [SupportedChainId.POLYGON_MUMBAI]: 'polygon_mumbai',
-  [SupportedChainId.CELO]: 'celo',
-  [SupportedChainId.CELO_ALFAJORES]: 'celo_alfajores',
-  [SupportedChainId.ARBITRUM_ONE]: 'arbitrum',
-  [SupportedChainId.ARBITRUM_GOERLI]: 'arbitrum_goerli',
-  [SupportedChainId.OPTIMISM]: 'optimism',
-  [SupportedChainId.OPTIMISM_GOERLI]: 'optimism_goerli',
-  [SupportedChainId.BNB]: 'bnb',
+  [ChainId.MAINNET]: 'mainnet',
+  [ChainId.GOERLI]: 'goerli',
+  [ChainId.SEPOLIA]: 'sepolia',
+  [ChainId.POLYGON]: 'polygon',
+  [ChainId.POLYGON_MUMBAI]: 'polygon_mumbai',
+  [ChainId.CELO]: 'celo',
+  [ChainId.CELO_ALFAJORES]: 'celo_alfajores',
+  [ChainId.ARBITRUM_ONE]: 'arbitrum',
+  [ChainId.ARBITRUM_GOERLI]: 'arbitrum_goerli',
+  [ChainId.OPTIMISM]: 'optimism',
+  [ChainId.OPTIMISM_GOERLI]: 'optimism_goerli',
+  [ChainId.BNB]: 'bnb',
+  [ChainId.AVALANCHE]: 'avalanche',
 }
 
-/**
- * Array of all the supported chain IDs
- */
-export const ALL_SUPPORTED_CHAIN_IDS: SupportedChainId[] = Object.values(SupportedChainId).filter(
-  (id) => typeof id === 'number'
-) as SupportedChainId[]
+export function isSupportedChain(chainId: number | null | undefined | ChainId): chainId is SupportedChainsType {
+  return !!chainId && SUPPORTED_CHAINS.indexOf(chainId) > 0
```

Comment:
This should probably just read `SUPPORTED_CHAINS.indexOf(chainId) !== -1`. `0` is a valid index.

Reply:
thanks for the catch